Megaparsec/MonadParsec.lean

```diff
@@ -36,7 +36,7 @@ On type variables:
 - `γ` is the type of the thing we're parsing out of the source.
 
 -/
-class MonadParsec (m : Type u → Type v) (℘ α E β : Type u) where
+class MonadParsec (m : Type u → Type v) (℘ α E β : Type u) [Ord β] [Ord E] where
```

Comment:
I disagree that these constraints belong here. Our typeclass is jackal as it is, and adding more hoops is kind of bad.

We can keep the constraint in the instances we provide and let the users come up with their own versions of deduplication, if they choose to accept that challenge.

(Sidenote: I still maintain that the pipe dream of extending MonadParsec by third-party users isn't much more than an overengineering. As I said in my analysis, if this polymorphism would've been actually needed, we wouldn't have had several different libraries defining their own typeclass in Haskell ecosystem, but rather all the Haskell parsecs would have been libraries depending on the original parsec).